### ISSUE TITLE
:tools: usage results in NoMethod Error

### ISSUE THREAD

**Issue by wolf99** [ThrowTheSwitch/Ceedling#107]
Using format as in example YAML file `:tools_test_compiler:` works fine.

But the option is documented in documentation *and* YAML comments as 
```
:tools:
  :test_compiler:
```
 This latter usage results in the following error if running rake task (e.g. `$ rake test`)
```
rake aborted!
NoMethodError: undefined method `strip!' for nil:NilClass
/mnt/c/project/test/vendor/ceedling/lib/ceedling/file_path_utils.rb:24:in `standardize'
/mnt/c/project/test/vendor/ceedling/lib/ceedling/configurator.rb:250:in `block in standardize_paths'
/mnt/c/project/test/vendor/ceedling/lib/ceedling/configurator.rb:250:in `each_pair'
/mnt/c/project/test/vendor/ceedling/lib/ceedling/configurator.rb:250:in `standardize_paths'
/mnt/c/project/test/vendor/ceedling/lib/ceedling/setupinator.rb:30:in `do_setup'
vendor/ceedling/lib/../lib/ceedling/rakefile.rb:46:in `<top (required)>'
vendor/ceedling/lib/ceedling.rb:66:in `load'
vendor/ceedling/lib/ceedling.rb:66:in `load_project'
/mnt/c/project/test/rakefile.rb:4:in `<top (required)>'
(See full trace by running task with --trace)
```

**Note:** Ruby 2.3.1p112 in rbenv 1.0.0-33

**Comment by mvandervoord** [comment]
Can you show an example of something that works and that doesn't? I'm not sure I am following.

**Comment by wolf99** [comment]
The following fails with the error initially reported
```
:tools:
  :test_compiler:
    :arguments:
      - '-Wall'
```

The following works

```
:tools_test_compiler:
  :arguments:
    - '-Wall'
```

**Comment by mvandervoord** [comment]
Thanks :)

**Comment by mvandervoord** [comment]
Ah, I see. It's assuming that if you are going to specify a test compiler, it was waiting for the executable when done this way. Since you only provided arguments, it failed out. I think I can protect against that. :)

**Comment by mvandervoord** [comment]
Fixed in #c6a09ca
